CMakeLists.txt

```diff
@@ -23,6 +23,11 @@ function(packageProject)
     set(PROJECT_COMPATIBILITY AnyNewerVersion)
   endif()
 
+  # we wanto to automatically add :: to our namespace, so only append if a namespace was given in the first place
+  if (DEFINED PROJECT_LIBRARY_NAMESPACE)
+    set(PROJECT_LIBRARY_NAMESPACE ${PROJECT_LIBRARY_NAMESPACE}::)
+  endif()
+
```

Comment:
Should we also create an alias target here for compatibility between using local and system version of the library? e.g.

```cmake
add_library(${LIBRARY_NAMESPACE}::${PROJECT_NAME} alias ${PROJECT_NAME})
```

That way it wouldn't matter to the user if the library was included via `add_subdirectory` or `find_package`.

Reply:
I don't follow... there is an alias in the `namespaced_dependency`'s `CMakeLists.txt`, did you by any chance mean to comment on that one?
I added the `if` in this code, because giving the parameter `INSTALL ${PROJECT_NAMESPACE}` to the `install` command makes it, so that the resulting name is `nsnamespaced_dependency` (missing the `::`) instead of `ns::namespaced_dependency`

Reply:
The idea was that PacakgeProject could take care of creating the alias for the library creator, as it already has been provided all necessary information. In that case you could remove the manual alias from the `namespaces_dependency`'s `CMakeLists.txt`.  For one it would remove potential code duplication and also guarantee that the local and system alias targets have the same name.

Reply:
I see. Sounds like a neat idea.